File: src/features/typing/data/go/minimum_size_subarray_sum.go
package main

import "math"

func minSubArrayLen(target int, nums []int) int {
	best := math.MaxInt
	left, sum := 0, 0
	for right, num := range nums {
		sum += num
		for sum >= target {
			if right-left+1 < best {
				best = right - left + 1
			}
			sum -= nums[left]
			left++
		}
	}
	if best == math.MaxInt {
		return 0
	}
	return best
}

File: src/features/typing/data/dart/minimum_size_subarray_sum.dart
class Solution {
  int minSubArrayLen(int target, List<int> nums) {
    var best = nums.length + 1;
    var left = 0;
    var sum = 0;
    for (var right = 0; right < nums.length; right++) {
      sum += nums[right];
      while (sum >= target) {
        if (right - left + 1 < best) {
          best = right - left + 1;
        }
        sum -= nums[left];
        left++;
      }
    }
    return best > nums.length ? 0 : best;
  }
}
